libraries/eosiolib/core/eosio/crypto_bls_ext.hpp

```diff
@@ -53,81 +53,81 @@ namespace eosio {
 
     inline int32_t bls_g1_add(const bls_g1& op1, const bls_g1& op2, bls_g1& res) {
         return internal_use_do_not_use::bls_g1_add(
-            op1.data(), sizeof(bls_g1),
-            op2.data(), sizeof(bls_g1),
-            res.data(), sizeof(bls_g1)
+            op1.data(), op1.size(),
+            op2.data(), op2.size(),
+            res.data(), res.size()
         );
     }
 
     inline int32_t bls_g2_add(const bls_g2& op1, const bls_g2& op2, bls_g2& res) {
         return internal_use_do_not_use::bls_g2_add(
-            op1.data(), sizeof(bls_g2),
-            op2.data(), sizeof(bls_g2),
-            res.data(), sizeof(bls_g2)
+            op1.data(), op1.size(),
+            op2.data(), op2.size(),
+            res.data(), res.size()
         );
     }
 
     inline int32_t bls_g1_weighted_sum(const bls_g1 g1_points[], const bls_scalar scalars[], uint32_t num, bls_g1& res) {
         return internal_use_do_not_use::bls_g1_weighted_sum(
-            num ? g1_points[0].data() : nullptr, num * sizeof(bls_g1),
-            num ? scalars[0].data() : nullptr, num * sizeof(bls_scalar),
+            num ? g1_points[0].data() : nullptr, num * g1_points[0].size(),
```

Comment:
Maybe it would be cleaner like this:
```c++
    inline int32_t bls_g1_weighted_sum(const bls_g1 g1_points[], const bls_scalar scalars[], uint32_t num, bls_g1& res)
    {
        if (num)
            return internal_use_do_not_use::bls_g1_weighted_sum(g1_points[0].data(), num * g1_points[0].size(),
                                                                scalars[0].data(),   num * scalars[0].size(),
                                                                num,
                                                                res.data(), res.size());
        return internal_use_do_not_use::bls_g1_weighted_sum(nullptr, 0, nullptr, 0, 0, res.data(), res.size());
    }
```

Reply:
My thoughts exactly. Done.